go/signedexchange/verifier.go

```diff
@@ -162,9 +164,14 @@ func (e *Exchange) Verify(verificationTime time.Time, certFetcher CertFetcher, l
 		// TODO: Implement Step 6 and 7 (certificate verification).
 
 		// Step 8: "Return "valid"."
-		return true
+
+		// Make sure to return non-nil value, even if the payload is empty.
+		if decodedPayload == nil {
+			decodedPayload = make([]byte, 0)
+		}
+		return decodedPayload
```

Comment:
Ah, I'm really sorry but I missed the case when this function succeeds and the result is empty. Then `Verify` should return a boolean value indicating success. (Using a difference between a nil slice and 0-length slice would be confusing)

Then the signature of this function would be

```go
func (e *Exchange) Verify(verificationTime time.Time, certFetcher CertFetcher, l *log.Logger) ([]byte, bool)
```

Reply:
No problem, done :)